client/pedersen_ec.go

```diff
@@ -32,10 +44,47 @@ func (c *Client) PedersenEC(val *big.Int) error {
 		return err
 	}
 
-	decommitVal, r := c.handler.pedersenECCommitter.GetDecommitMsg()
+	decommitVal, r := c.committer.GetDecommitMsg()
 	if err = c.decommit(decommitVal, r); err != nil {
 		return err
 	}
 
+	if err := c.close(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *PedersenECClient) getH() (*pb.ECGroupElement, error) {
+	initMsg := &pb.Message{
+		ClientId: c.id,
+		Schema:   pb.SchemaType_PEDERSEN_EC,
+		Content:  &pb.Message_Empty{&pb.EmptyMsg{}},
+	}
+	if err := c.send(initMsg); err != nil {
+		return nil, err
+	}
+
+	resp, err := c.receive()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I thought about this even before refactoring, but somehow I didn't remember implementing it now. I like your suggestion, and I really like the resulting simplifications in protocol client's function that do this bouncing! I decided not to name the new bouncing function `bounce`, but `getResponseTo`. So this is what we have in our functions now:

```go
message := &pb.Message{...} // form the message
response, err := c.getResponseTo(message) // instead of c.recv()
if err != nil {
	return false, err
}
// do something with response
```